### PR TITLE
add new strategy `delete_all_using_limit_in_batches`

### PR DESCRIPTION
**Pull request by LeoMo42**
add `order` for `with tenant_id_field` case as it force DB using needed DB index

### CODE REVIEW

**Comment by LeoMo42** [comment]
@Azdaroth , could you please review?

File: lib/tartarus/archive_strategy.rb [hunk 1]
```diff
@@ -1,5 +1,5 @@
 class Tartarus::ArchiveStrategy
-  def for(strategy_name)
+  def for(strategy_name, batch_size)
```
Comment (Azdaroth) [inline lib/tartarus/archive_strategy.rb]: can we also make it a keyword argument for a batch_size? Maye having 0 as the default would be helpful here as somehow now, everyone has to pass batch_size even for strategies that don't use this argument

**Review by Azdaroth — COMMENTED** [review]
Looks great!

Just one minor comment about batch_size argument for ArchiveStrategy.

Also, can you please add the change to Changelog? And then we will be able to release 0.3.0 💪 let me know on Slack what email you want to have as the co-owner of the gem, so you will be able to release it as well.

**Comment by LeoMo42** [comment]
@Azdaroth , could you please review again?

**Review by Azdaroth — APPROVED** [review]
💪 🐉 Opa!

**Comment by Azdaroth** [comment]
@LeoMo42 0.3.0 is released 🎉

**Pull request merged**